hypothesis-python/src/hypothesis/internal/conjecture/providers.py

```diff
@@ -74,16 +74,34 @@
 COLLECTION_DEFAULT_MAX_SIZE = 10**10  # "arbitrarily large"
 
 
-# The available `PrimitiveProvider`s, and therefore also the available backends
-# for use by @settings(backend=...). The key is the name to be used in the backend=
-# value, and the value is the importable path to a subclass of PrimitiveProvider.
-#
-# See also
-# https://hypothesis.readthedocs.io/en/latest/strategies.html#alternative-backends-for-hypothesis.
-#
-# NOTE: the PrimitiveProvider interface is not yet stable. We may continue to
-# make breaking changes to it. (but if you want to experiment and don't mind
-# breakage, here you go!)
+#: Registered Hypothesis backends. This is a dictionary whose keys are the name
+#: to be used in |settings.backend|, and whose values are a string of the absolute
+#: importable path to a subclass of |PrimitiveProvider|, which Hypothesis will
+#: instantiate when your backend is requested by a test's |settings.backend| value.
+#:
+#: For example, the default Hypothesis backend is registered as:
+#:
+#: .. code-block:: python
+#:
+#:    from hypothesis.internal.conjecture.providers import AVAILABLE_PROVIDERS
+#:
+#:    AVAILABLE_PROVIDERS["hypothesis"] = "hypothesis.internal.conjecture.providers.HypothesisProvider"
```

Comment:
I think we want to call out that the purpose of this design is that you can ship a Hypothesis plugin which will register your alternative backend _without importing your library_ unless and until the user specifically requests that backend.  This is vital for fast test startup times.

(meta / later: maybe we should measure and report plugin import times?)